### PR TITLE
chore: CircleCI support

### PR DESCRIPTION
**Pull request by malept**
* [x] I have read the [contribution documentation](https://github.com/electron/electron-packager/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] The changes are appropriately documented (if applicable).
* [x] The changes have sufficient test coverage (if applicable).
* [x] The testsuite passes successfully on my local machine (if applicable).

**Summarize your changes:**

Move Linux/macOS CI builds to CircleCI, now that we're in the Electron org.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    - run: npm install
+    - *step-restore-electron-cache
+    - run: test/ci/_before_script.js
+    - *step-save-electron-cache
+    - run: npm test
+    - run: npm run codecov
+
+version: 2
+jobs:
+  test-linux-8:
```
Comment (MarshallOfSound) [inline .circleci/config.yml]: Should we add `10` in here as well?

**Comment by MarshallOfSound** [comment]
We should probs disable travis / appveyor (one of them)

File: .circleci/config.yml (hunk 1 above)
Comment (malept) [reply .circleci/config.yml]: Sure, why not.

**Comment by malept** [comment]
Travis is slower, I'll remove that.

**Pull request merged**